packages/fxa-content-server/app/scripts/lib/sentry.js

```diff
@@ -1,233 +0,0 @@
-      SAME_ERROR_ATTEMPTS = 0;
-    }
-    if (SAME_ERROR_LIMIT === SAME_ERROR_ATTEMPTS) {
-      return false;
-    }
-    PREVIOUS_ERROR = data.message;
-    return true;
-  }
-
-  return true;
-}
-
-/**
- * Overwrites sensitive query parameters with a dummy value.
- *
- * @param {String} url
- * @returns {String} url
- * @private
- */
-function cleanUpQueryParam(url = '') {
-  var startOfParams = url.indexOf('?');
-  var newUrl = url;
-  var params;
-
-  if (startOfParams === -1) {
-    return newUrl;
-  }
-
-  params = Url.searchParams(url.substring(startOfParams + 1));
-  newUrl = url.substring(0, startOfParams);
-
-  if (_.isObject(params)) {
-    Object.keys(params).forEach(function(key) {
-      // if the param is a PII (not allowed) then reset the value.
-      if (!_.contains(ALLOWED_QUERY_PARAMETERS, key)) {
-        params[key] = 'VALUE';
-      }
-    });
-
-    newUrl += Url.objToSearchString(params);
-  }
-
-  return newUrl;
-}
-
-/**
- * Creates a SentryMetrics object that starts up Raven.js
- *
- * Read more at https://github.com/getsentry/raven-js
- *
- * @param {String} host
- * @param {String} [release] - content server release version
- * @constructor
- */
-function SentryMetrics(host, release) {
-  this._logger = new Logger();
-  this._release = release;
-
-  if (host) {
-    // use __API_KEY__ instead of the real API key because raven.js requires it
-    // we can configure the real API key on the server using our local endpoint instead.
-    // See issue https://github.com/getsentry/raven-js/issues/346 for more information
-
-    this._endpoint = '//__API_KEY__@' + host + '/metrics-errors';
-  } else {
-    this._logger.error('No Sentry host provided');
-    return;
-  }
-
-  try {
-    Raven.config(this._endpoint, this._ravenOpts).install();
-    Raven.debug = false;
-  } catch (e) {
-    Raven.uninstall();
-    this._logger.error(e);
-  }
-}
-
-SentryMetrics.prototype = {
-  /**
-   * Specialized raven.js endpoint string
-   *
-   * See https://raven-js.readthedocs.org/en/latest/config/index.html#configuration
-   */
-  _endpoint: null,
-  /**
-   * raven.js settings
-   *
-   * See https://raven-js.readthedocs.org/en/latest/config/index.html#optional-settings
-   */
-  _ravenOpts: {
-    dataCallback: beforeSend,
-    shouldSendCallback: shouldSendCallback,
-  },
-
-  /**
-   * Exception fields that are imported as tags
-   */
-  _exceptionTags: ['code', 'context', 'errno', 'namespace', 'status'],
-
-  /**
-   * Capture an exception. Error fields listed in _exceptionTags
-   * will be added as tags to the raven data.
-   *
-   * @param {Error} err
-   */
-  captureException(err) {
-    var tags = {};
-
-    this._exceptionTags.forEach(function(tagName) {
-      if (tagName in err) {
-        tags[tagName] = err[tagName];
-      }
-    });
-
-    var extraContext = {
-      tags: tags,
-    };
-
-    if (this._release) {
-      // add release version if available
-      extraContext.release = this._release;
-    }
```

Comment:
I don't understand why this module is being completely removed.  Wouldn't we lose the extra context at the very least?

Reply:
We are now using the direct sentry package instead of a fork of the outdated package. We no longer send the metrics to a custom endpoint proxy. We should make sure that we don't leave anything behind in the new version.

Reply:
@chenba After looking into this a bit more, I think we should go ahead with this conversion to get us off the the legacy raven package, and file a follow up issue to add any extra info after we see what info we are getting in sentry.

Reply:
This file was a wrapper.  I don't know much about the Sentry API and how it has changed from the old one to new one, so I can't speak to much work it is to re-wrap the new one.  But it doesn't seem possible that, by default, the new Sentry package will handle everything in here.  For example, there's a hard coded whitelist of query params (https://github.com/mozilla/fxa/pull/2959/files/31407a79ef0f1a0fc9318607457a0b242207c393#diff-af20f0c892e15cdf0f5aa74531981dcbL10).  Without that or any additional configuration, I don't see how Sentry would know what query params are allow to be collected.

Reply:
I added this file when revamping the Sentry support in auth-server to do filtering of request data:
https://github.com/mozilla/fxa/pull/3128/files#diff-a44a71b1f6c14d7564bb44a782497ff4

I assume some aspects of this can be applied here.